beaconChain/typescript/interfaces/state.ts

```diff
@@ -0,0 +1,105 @@
+// TODO replace uint, hash32, bytes
+
+// These interfaces relate to the data structures for beacon chain state
+
+import { AttestationRecord } from "./blocks";
+
+type uint24 = number;
+type uint64 = number;
+type uint384 = number;
+type hash32 = number;
+
+export interface BeaconState {
+  // Slot of last validator set change
+  validatorSetChangeSlot: uint64,
+  // List of validators
+  validators: ValidatorRecord[],
+  // Most recent crosslink for each shard
+  crosslinks: CrosslinkRecord[],
+  // Last cycle-boundary state recalculation
+  lastStateRecalculationSlot: uint64,
+  // Last finalized slot
+  lastFinalizedSlot: uint64,
+  // Last justified slot
+  lastJustifiedSlot: uint64,
+  // Number of consecutive justified slots
+  justifiedStreak: uint64,
+  // Committee members and their assigned shard, per slot
+  shardAndCommitteeForSlots: [[ShardAndCommittee]],
```

Comment:
Please update all the arrays to use consistent notations (ie. type[] or type[][])